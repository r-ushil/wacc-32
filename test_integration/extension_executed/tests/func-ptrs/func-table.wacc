

# Output:
# 4

begin
  int one(int n) is return n + 1 end
  int two(int n) is return n + 2 end
  int three(int n) is return n + 3 end

  int(int)[] fns = [one, two, three];

  # two plus two is four!
  int four = call fns[1](2);

  println four
end
