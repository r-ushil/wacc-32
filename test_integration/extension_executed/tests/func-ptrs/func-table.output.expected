Compiler exit code: 0
--------- stdout ---------
4
--------------------------
Exit code: 0
